src/test/java/com/jcabi/log/ParseLevelInformationTest.java

```diff
@@ -0,0 +1,105 @@
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.log;
+
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * ParseLevelInformationTest test case.
+ * @author Jose V. Dal Pra Junior (jrdalpra@gmail.com)
+ * @version $Id$
+ * @since 0.18
+ */
+public class ParseLevelInformationTest {
+
+    /**
+     * Wrong information to be parsed.
+     */
+    private static final String WRONG_INFORMATION = "INFO;10,WARN;32";
+
+    /**
+     * ParseLevelInformation can parse the information correctly when it's
+     * with the right pattern.
+     */
+    @Test
+    @SuppressWarnings("PMD.UseConcurrentHashMap")
+    public final void parsesCorrectlyTheInformation() {
+        final Map<String, String> parsed = new ParseLevelInformation(
+            "INFO:2;10,WARN:2;32"
+        ).parse();
+        Assert.assertThat(parsed, Matchers.hasEntry("INFO", "2;10"));
+        Assert.assertThat(parsed, Matchers.hasEntry("WARN", "2;32"));
+    }
+
+    /**
+     * ParseLevelInformation can throw an exception when information is
+     * not with the right pattern.
+     */
+    @Test
+    public final void throwsAnExceptionWhenParsingIncorrectInformation() {
+        try {
+            new ParseLevelInformation(WRONG_INFORMATION).parse();
+            Assert.fail();
+        } catch (final IllegalStateException ex) {
+            Assert.assertThat(
+                ex.getMessage(), Matchers.equalTo(
+                    String.format(
+                        StringUtils.join(
+                            "Information is not using the pattern ",
+                            "KEY1:VALUE,KEY2:VALUE %s"
+                        ), WRONG_INFORMATION
+                    )
+                )
+            );
+        }
+    }
+
+    /**
+     * ParseLevelInformation can throw an exception when passing information
+     * with a wrong type of level.
+     */
+    @Test
+    public final void throwsAnExceptionWhenParsingWrongLevelType() {
+        try {
+            new ParseLevelInformation("INFO:2;10,EXTREME:2;32").parse();
+            Assert.fail();
+        } catch (final IllegalStateException ex) {
+            Assert.assertThat(
+                ex.getMessage(),
+                Matchers.equalToIgnoringCase("Unknown level 'EXTREME'")
```

Comment:
@jrdalpra do you want to keep `equalToIgnoringCase` here? Can you explain me advantages of this approach?